### PR TITLE
Added user_categories migration, model, endpoint

### PR DESCRIPTION
**Pull request by green-arrow**
Addresses #39

### CODE REVIEW

File: priv/repo/migrations/20160822020401_create_user_category.exs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  def change do
+    create table(:user_categories) do
+      add :user_id, references(:users, on_delete: :nothing)
+      add :category_id, references(:categories, on_delete: :nothing)
+
+      timestamps()
+    end
+
+    create index(:user_categories, [:user_id, :category_id], unique: true, name: :index_projects_on_user_id_category_id)
```
Comment (joshsmith) [inline priv/repo/migrations/20160822020401_create_user_category.exs]: Same point on name here as before.

File: test/controllers/user_category_controller_test.exs [hunk 2]
```diff
@@ -0,0 +1,112 @@
+defmodule CodeCorps.UserCategoryControllerTest do
+  use CodeCorps.ConnCase
+
+  alias CodeCorps.UserCategory
+  alias CodeCorps.Repo
```
Comment (joshsmith) [inline test/controllers/user_category_controller_test.exs:4]: Would reorder alpha.

**Comment by joshsmith** [comment]
Same point here on a `UserCategoryViewTest`.

I would also consider doing a `UserViewTest` and ensuring that the `has_many` categories through is present there.

Otherwise looks great!

**Comment by green-arrow** [comment]
@JoshSmith - If we can get the user roles branch merged, I'll have a basis to add on to the `UserViewTest` I added there :)

**Comment by green-arrow** [comment]
@JoshSmith - Addressed the code review comments. My computer is getting ready to die and I have no idea where my charger is, so feel free to squash and merge (have to hit `Merge pull request` first to get that option) if there are no outstanding issues. If there are, I'll make sure to get them when I wake up in the morning and find my charger :)

**Comment by joshsmith** [comment]
👍 Looks great, thanks @green-arrow!

**Pull request merged**